Added a vm_extra_config option for the folder the VM should be created in

stewrutledge opened the pull request:
Added a section in vm_extra_config (folder) which allows you to specify which vm folder the new guest should be created in.

First there is a check that the folder exists on the system, failing if it doesn't.

If it does exist, the vmfmor variable is set to the first key (I am not sure how this behaves if there are multiple folders with the same name) found in the dictionary returned by _get_managed_objects.

If it is not set, it just goes to the current default (the root folder on the datacenter/host).

```
    args:
        vm_extra_config:
          vcpu.hotadd: yes
          mem.hotadd:  yes
          notes: This is a test VM
          folder: AutoDeploy
```

sterutkb committed:
Added a vm_extra_config option for the folder the VM should be created in

sterutkb committed:
Added a vm_extra_config option for the folder the VM should be created in

mpdehaan commented:
This seems reasonable but I'd like to see the new option referenced in examples in the DOCUMENTATION string, else folks aren't going to find out that this option exists.

Can you update the DOCUMENTATION part above in the same file to show an example?

sterutkb committed:
Added a line in the documentation showing the folder option and updated error message to match the others (cannot find vs could not find)

sterutkb committed:
Added a line in the documentation showing the folder option and updated error message to match the others (cannot find vs could not find)

stewrutledge commented:
Updated the documentation to show the option under extra_config.

I think this module could use a bit more documentation in general, that'll have to be another pull request in the future.

gnosek committed:
Fix exponential regex performance issue

filter_leading_non_json_lines effectively does

re.match(".*\w+=\w+.*", line)

for every line of output. This has abysmal performance in case of large
Base64-encoded data (which ultimately does not match the regex but does
match the .*\w+= part) as returned e.g. by the template module (diffs).

Replacing the match with

re.search("\w=\w", line)

drops the complexity back to linear, and actually usable with large
diffs from the template module (a 150 KB Base64 diff kept Ansible
spinning at 100% cpu for minutes).

Also, check the easy cases (line.startswith) first while we're here.

Closes: #8932

lukesneeringer committed:
Envirionment variable support in GCE inventory plugin.

Jmainguy committed:
Fixes issue #8339 where Scientific Linux incorrectly tries to use FedoraStrategy of hostnamectl

ebargtuo committed:
Fix a typo in gem module documentation

Commit:
Merge branch 'bug8392' of git://github.com/gnosek/ansible into devel

jimi-c committed:
Fix literal block multiline parsing

Fixes #8394

Commit:
Add some whitespace to make paramiko_ssh more browseable

jimi-c committed:
Remove unnecessary unicode conversions from parse_kv

Fixes #8425

Commit:
Use atomic move function on known host file in paramiko to prevent rare occurance of Control-C
during file operations.

Commit:
Revert ""allow_no_value=True" for ini_file module so /etc/my.cnf can be read without a problem"

This reverts commit cbdc9f1153e665fe0e62a11ed7c244adb83ecbd9.

Code needs to make allowances for particular python versions and try/except
around this block.

jlaska committed:
Require python-crypto >= 2.6 for debian package builds

The ansible PyPi package, and RPM both require crypto>=2.6.  This change
carries that dependency into the ansible DEB package.  Note, for
ubuntu-12.04, this dependency is only satisfied by a PPA.  Users
installing from 'ppa:ansible/ansible' will not be affected..

Commit:
Revert "Use atomic move function on known host file in paramiko to prevent rare occurance of Control-C"

This reverts commit 642b183fb6b0632167130774c2ebfc44d88ac278.

jlaska committed:
Merge pull request #8426 from jlaska/update_integration_Makefile

Include unicode and var_precedence tests in 'all' target

jimi-c committed:
Atomically move known hosts file into place when edited by paramiko connections

Fixes #8169

mpdehaan committed:
Merge pull request #8409 from Jmainguy/hostname_8339

Fixes issue #8339 where Scientific Linux incorrectly tries to use Fedora...

mpdehaan committed:
Merge pull request #8415 from ebargtuo/patch-1

Fix a typo in gem module documentation

mpdehaan committed:
Merge pull request #8402 from lukesneeringer/gce-inventory-envvars

Envirionment variable support in GCE inventory plugin.

stewrutledge committed:
Merge branch 'devel' of github.com:stewrutledge/ansible into devel

stewrutledge commented:
Closing as this was made on my local devel branch.

Re-opened with own branch here:
https://github.com/ansible/ansible/pull/8451
